projects/packages/videopress/src/client/state/actions.js

```diff
@@ -228,6 +231,47 @@ const setPurchases = purchases => {
 	return { type: SET_PURCHASES, purchases };
 };
 
+const updateVideoPoster = ( id, guid, data ) => async ( { dispatch } ) => {
+	const path = `${ WP_REST_API_VIDEOPRESS_ENDPOINT }/${ guid }/poster`;
+
+	const pollPoster = ( resolve, reject ) => {
+		setTimeout( async () => {
+			try {
+				const resp = await apiFetch( { path, method: 'GET' } );
+
+				if ( resp?.data?.generating ) {
+					pollPoster( resolve, reject );
+				} else {
+					resolve( resp?.data?.poster );
+				}
+			} catch ( error ) {
+				reject( error );
+			}
+		}, 2000 );
+	};
+
+	try {
+		dispatch( { type: SET_UPDATING_VIDEO_POSTER, id } );
+
+		const resp = await apiFetch( { method: 'POST', path, data } );
+		let poster;
+
+		if ( resp?.data?.generating ) {
+			// Poll the poster image until generated
+			poster = await new Promise( ( resolve, reject ) => {
```

Comment:
Don't think we should keep think waiting until we finish the pooling, that's the reason we have `dispatch` available, let's continue the flow and dispatch as soon it got resolved directly at pooling function.